vtex/faststore
# Add option to enable brotli nginx plugin

## What's the purpose of this pull request?
Allow the nginx plugin to enable brotli encoding in the generated configuration.

ngx_brotli -> https://github.com/google/ngx_brotli

## How it works? 

- Set `NGINX_BROTLI` env var to true when building the gatsby site
- A set of directives enabling brotli compression will be added to the generated config
- Launch nginx with the generated config - the nginx must have the brotli plugin installed, otherwise launch will fail (this is the reason why the feature is opt-in)
- The response to any requests which `Accept-Encoding` header contains `br` will be brotli-encoded.

## How to test it?
see previous topic and then one might use the following nginx image https://hub.docker.com/r/fholzer/nginx-brotli/ which has brotli installed statically instead of the original docker image in the `docker:serve` npm script.

## References
